### PR TITLE
send host header for remote

### PR DESCRIPTION
**Pull request by tr4n2uil**
- [X ] By placing an `X` in the preceding checkbox, I verify that I have signed the [Contributor License Agreement](https://github.com/SeleniumHQ/selenium/blob/master/CONTRIBUTING.md#step-6-sign-the-cla)

Host header important for remote selenium grid using traffic accelerators like CloudFlare like https://hub-cloud.browserstack.com

---
This change is [Reviewable](https://reviewable.io/reviews/seleniumhq/selenium/2083)

### CODE REVIEW

**Comment by lukeis** [comment]
In general I'm :+1:  can you add a test for it?

**Comment by tr4n2uil** [comment]
Thanks, writing the test case 👍🏻

**Comment by tr4n2uil** [comment]
@lukeis Please check the test cases added.

File: py/test/selenium/test_remote_connection.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        self.rc = remote_connection.RemoteConnection(REMOTE_URL)
+        self.rc.execute(Command.STATUS, {})
+        self.mock_add_header.assert_called_with('Host', REMOTE_HOST)
+
+    def test_host_header_with_keep_alive(self):
+        self.rc = remote_connection.RemoteConnection(REMOTE_URL, keep_alive=True)
+        self.rc.execute(Command.STATUS, {})
+        self.mock_request.assert_called_with('GET', '/wd/hub/status', None, {
+            'GET': '/wd/hub/status', 
+            'Host': 'grid.seleniumhq.org',
```
Comment (lukeis) [inline py/test/selenium/test_remote_connection.py]: use the REMOTE_HOST variable here instead

File: py/test/selenium/test_remote_connection.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
+# Licensed to the Software Freedom Conservancy (SFC) under one
```
Comment (lukeis) [inline py/test/selenium/test_remote_connection.py]: I don't think this is the correct location for this test... other than moving into webdriver, we'd also need to make sure the test is actually run.

@davehunt

**Comment by tr4n2uil** [comment]
@lukeis used REMOTE_HOST and added the test to selenium_test_suite. Please check.

File: py/test/selenium/selenium_test_suite.py [hunk 3]
```diff
@@ -29,6 +30,7 @@ def suite():
         unittest.makeSuite(test_default_server.TestDefaultServer),
         unittest.makeSuite(test_google.TestGoogle),
         unittest.makeSuite(test_i18n.TestI18n),
+        unittest.makeSuite(test_remote_connection.TestRemoteConnection),
```
Comment (lukeis) [inline py/test/selenium/selenium_test_suite.py]: this test suite isn't really used...

The python tests are currently built / run with selenium's custom build tool (crazy-fun) and here's the main definition for python:

https://github.com/SeleniumHQ/selenium/blob/master/py/build.desc

Basically to get a test to run, you'll need to add the test case into https://github.com/SeleniumHQ/selenium/tree/master/py/test/selenium/webdriver/common

But those all assume you're running with a particular browser... which for this we don't really need and you did the correct thing by making a unit test... which we don't really have a structure for running...

@davehunt @AutomatedTester halp?

Comment (tr4n2uil) [inline py/test/selenium/selenium_test_suite.py]: @lukeis Cool thanks! 
@davehunt @AutomatedTester Please help, will make the changes accordingly.

Comment (davehunt) [inline py/test/selenium/selenium_test_suite.py]: /me defers to @AutomatedTester as I'm only just learning how the Python tests are run

Comment (tr4n2uil) [inline py/test/selenium/selenium_test_suite.py]: @AutomatedTester Any suggestions ?

Comment (tr4n2uil) [inline py/test/selenium/selenium_test_suite.py]: @AutomatedTester @lukeis Do we have any update here ? Can we expedite this, our production build is waiting for this change.

Comment (AutomatedTester) [inline py/test/selenium/selenium_test_suite.py]: As @lukeis said, this isnt used, you need to create a remote directory in https://github.com/SeleniumHQ/selenium/tree/master/py/test/selenium/webdriver and add the test file there.

**Comment by tr4n2uil** [comment]
@AutomatedTester Thanks for the note. I have made the necessary changes. Kindly check. 
cc @lukeis

**Comment by tr4n2uil** [comment]
@lukeis Am sorry to ask again, but would it be possible to expedite this one ? It would be really helpful for us get this done for our production optimisations.

**Comment by tr4n2uil** [comment]
@lukeis Any update here ?

**Comment by lukeis** [comment]
I like the test you added, but it's not currently run with the test suite. I'm kind of waiting for @davehunt to do something he planned with the python test suite before attempting to give you direction on how to better add the test so that it is actually run

**Comment by tr4n2uil** [comment]
Looks promising, am waiting :)

**Comment by davehunt** [comment]
@lukeis I'm not sure when that will be ready, so I think this can probably land as it is. The test should run as part of the command: `./go //py:remote_firefox_test:run`. The problem is that these tests are not currently being run as part of CI or the release process. I'm hoping to address that eventually, but adding one more test at this time would not be an issue.

**Comment by tr4n2uil** [comment]
@lukeis @davehunt Pushed the rename of file to make the test run as part of `./go //py:remote_firefox_test:run`

Can this be merged ?

**Comment by AutomatedTester** [comment]
Closing as we have moved away from this model of requests. If we still want it please raise a bug and let me know

**Pull request closed without merging**